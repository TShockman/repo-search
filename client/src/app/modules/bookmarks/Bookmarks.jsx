import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import RepoList from '../common/repo-list';

import './bookmarks.scss';

export default class Bookmarks extends PureComponent {
  static propTypes = {
    repos: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.number.isRequired,
          full_name: PropTypes.string.isRequired,
          stargazers_count: PropTypes.number.isRequired,
          forks_count: PropTypes.number.isRequired,
          html_url: PropTypes.string
        })).isRequired,
    getBookmarks: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.getBookmarks();
  }

  handleRefresh = event => {
    event.stopPropagation();
    this.props.getBookmarks();
  };

  render() {
    const {repos} = this.props;
    return (
        <section>
          <div className="bookmarksHead">
            <h2>Bookmarks</h2>
            <button onClick={this.handleRefresh}>Refresh</button>
          </div>
          <RepoList repos={repos}/>
        </section>
    )
  }
}