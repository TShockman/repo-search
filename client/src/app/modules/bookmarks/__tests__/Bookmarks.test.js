import React from 'react';
import { mount } from 'enzyme';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Bookmarks from '../Bookmarks';

const renderer = new ReactShallowRenderer();

const getProps = () => ({
  getBookmarks: jest.fn(),
  repos: [
    {
      id: 123,
      full_name: 'test/mock',
      stargazers_count: 444,
      forks_count: 1,
      html_url: 'http://fake.io'
    },
    {
      id: 456,
      full_name: 'test2/mock2',
      stargazers_count: 4444,
      forks_count: 2,
      html_url: 'http://fake.com'
    },
    {
      id: 789,
      full_name: 'test3/mock3',
      stargazers_count: 44444,
      forks_count: 3,
      html_url: 'http://fake.edu'
    }
  ]
});

describe('Bookmarks component', () => {
  it('renders consistently', () => {
    const props = getProps();
    expect(renderer.render(<Bookmarks {...props}/>)).toMatchSnapshot();
  });
  it('handles clicks on refresh', () => {
    const props = getProps();
    const component = new Bookmarks(props);
    const mockEvent = {
      stopPropagation: jest.fn()
    };
    component.handleRefresh(mockEvent);
    expect(mockEvent.stopPropagation).toHaveBeenCalledTimes(1);
    expect(props.getBookmarks).toHaveBeenCalledTimes(1);
  });
  it('gets bookmarks on mount', () => {
    const props = getProps();
    mount(<Bookmarks {...props}/>);
    expect(props.getBookmarks).toHaveBeenCalledTimes(1);
  });
});