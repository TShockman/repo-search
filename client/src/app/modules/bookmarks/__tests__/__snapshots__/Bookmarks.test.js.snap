// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarks component renders consistently 1`] = `
<section>
  <div
    className="bookmarksHead"
  >
    <h2>
      Bookmarks
    </h2>
    <button
      onClick={[Function]}
    >
      Refresh
    </button>
  </div>
  <RepoList
    repos={
      Array [
        Object {
          "forks_count": 1,
          "full_name": "test/mock",
          "html_url": "http://fake.io",
          "id": 123,
          "stargazers_count": 444,
        },
        Object {
          "forks_count": 2,
          "full_name": "test2/mock2",
          "html_url": "http://fake.com",
          "id": 456,
          "stargazers_count": 4444,
        },
        Object {
          "forks_count": 3,
          "full_name": "test3/mock3",
          "html_url": "http://fake.edu",
          "id": 789,
          "stargazers_count": 44444,
        },
      ]
    }
  />
</section>
`;
